my-sms/app/views/layouts/application.html.erb

```diff
@@ -1,14 +1,31 @@
 <!DOCTYPE html>
 <html>
-<head>
-  <title>MySms</title>
-  <%= stylesheet_link_tag    "application", :media => "all" %>
-  <%= javascript_include_tag "application" %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title>MySms</title>
+    <%= stylesheet_link_tag    "application", :media => "all" %>
+    <%= javascript_include_tag "application" %>
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+    <!-- display flashes -->
+    <% flash.each do |type, msg| %>
+      <div class="alert <%= get_alert_class(type) %>">
+        <!-- test if message is string or iterable -->
+        <% if msg.respond_to?(:each) %>
+          <ul class="alert-list">
+            <% msg.each do |msg| %>
+              <li class="alert-item"><%= msg %></li>
+            <% end %>
+          </ul>
+        <% else %>
+          <%= msg %>
+        <% end %>
+        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+    <% end %>
```

Comment:
There was probably a bit of confusion from my last comment, I meant to only show the the fact that there _is_ an error in the flash, and then have more error details displayed inline. Consider replacing this with the `<%= bootstrap_flash %>` I mentioned in slack?

Reply:
I got that, just took it as an exercise to dive into what flashes could be used for and investigate the process the rails controller goes through.

I looked more into the bootstrap_flash helper to try and implement it now and it is slightly incompatible. My implementation handles an array of errors and I'd rather keep that implementation than try and put formatting directly into the flash server side.